GDextension Windows fix

AverageNerdz opened the pull request:
This PR addresses and should fix issue #5467. It resolves the problem related to the gdextension compiling fail while using Windows.

Related Issues
I was unable to create a version that works seamlessly for both Linux and Windows in one go. However, I believe it can be addressed by adjusting the CMakeLists.txt to detect the operating system and configure accordingly. Testing across platforms is recommended to confirm everything works as expected.

Closes #5467

Progress Checklist

 PR author has checked that this PR works as intended and doesn't break existing features.
 Initial code review passed.
 Functionality is confirmed working by another person.
 Final code review is passed and code conforms to the [style guide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

hhyyrylainen reviewed (changes requested):
That's a lot of cmake changes. I hope most are unnecessary as I find quite many of them questionable or at least against my style of cmake I'm used to.

More critically with these changes the compiling the native libraries for export has some warnings:
```sh
$ dotnet run --project Scripts -- native Package
# I cut out a lot of extra text
CMake Warning:
  Manually-specified variables were not used by the project:

    THRIVE_GODOT_API_FILE
# This is significant as our scripts use Godot to generate a fresh set of binding definitions so if those aren't used the builds are incorrect... (instead they rely on the files in the godot-cpp repo that I guess could be used but for now I decided not to use them)
```

I was expecting the changes to cause the builds to fail, but apparently that wasn't the case, though I noticed this put the intermediate build artifacts into an extra lib folder which I don't find that useful to have (and so seems to be unnecessary forced configuration).

AverageNerdz commented:
Okay i was able to edit the main and the native files as they were the ones who had the most problems, here's a list of the edits i made:
1) Changed CMAKE_ARCHIVE_OUTPUT_DIRECTORY, CMAKE_LIBRARY_OUTPUT_DIRECTORY, and CMAKE_RUNTIME_OUTPUT_DIRECTORY to ${CMAKE_BINARY_DIR} in main. This removes the extra lib folder for build artifacts.
2) I added the option for THRIVE_GODOT_API_FILE in the main file however this option is still not being used in the project so to properly address this i'd need more infos/help to understand how to use this file in the appropriate place in the build system.
3)Removed global C++ standard settings from the main file. (3 and 4 tie in together)
4) Set C++ standard per target in native using set_target_properties.
5) Kept the static runtime library setting in the main i think that this solution might be discussed further as it might not be appropriate for a DLL to be loaded into the game process.
6) I have updated compile options in native to ensure NDEBUG is defined for both Release and Distribution.
7) Removed manual linking of libcmt and libcmtd in native.
8) I left the formatting of the precompiled headers as-is, as it has to be tested later with CLion further.
9) Removed the NODEFAULTLIB flags as they were pointed out to be unnecessary with the correct runtime library configuration.

AverageNerdz commented:
https://github.com/AverageNerdz/Thrive/tree/98bbb8c772dc6c26bb3ac726a1499b2b75d7eec8

hhyyrylainen commented:
I've made a fresh PR where this PR is used as the base but then I reverted most of the changes and only added back the minimal library changes that were necessary to get things compiling: https://github.com/Revolutionary-Games/Thrive/pull/5477

I'm not sure what that tree link above is for.

Also I noticed you didn't make the branch directly in the Thrive repo, you should do that in the future as that makes cooperation a lot easier to not have a branch in a fork.
